proxy/xml_validator.rb

```diff
@@ -0,0 +1,20 @@
+require 'nokogiri'
+
+module XMLValidator
+
+    def self.validate(xml)
+        xsd = Nokogiri::XML::Schema(File.read("post_joke.xsd"))
+        doc = Nokogiri::XML(xml)
+
+        errors = []
+        
+        xsd.validate(doc).each do |error|
+            errors.push(error.message)
+        end
+
+        return true if errors == []
```

Comment:
With this statement, your return value is either `true` or an array of errors.
A better approach would be either to throw all errors in a custom error message or to return a composite list, with of errors.

```suggestion
        return [errors.empty?, errors]
```
Later you can use this values to 
```ruby
is_valid, errors =  XMLValidator.validate(xml)
if !is_valid do 
    #treat errors
end
```